tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -534,4 +499,74 @@ void initTorchFunctions(PyObject* module) {
   }
 }
 
+PyObject* handle_torch_function(PythonArgs &r, PyObject* args, PyObject* kwargs, PyTypeObject &torch_api) {
+  PyObject* torch_api_function =
+    PyObject_FastGetAttrString((PyObject*)&torch_api, const_cast<char*>(r.get_func_name().data()));
+    PyObject* ret;
+    // there must be at least one overloaded argument at this point, since r.has_torch_function() is true
+    // so ret will never be returned unset
```

Comment:
Yes, invariants are nice, but you know what's even better? Checked invariants. Set `PyObject* ret = nullptr` so there isn't uninitialized memory, and `TORCH_INTERNAL_ASSERT` ret is non-null at the end.